src/main/java/com/easypost/model/PaginatedCollection.java

```diff
@@ -0,0 +1,70 @@
+package com.easypost.model;
+
+import com.easypost.exception.General.EndOfPaginationError;
+import lombok.Getter;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+@Getter
+public abstract class PaginatedCollection extends EasyPostResource {
+
+    private Boolean hasMore;
+
+    /**
+     * Get the next page of a paginated collection.
+     *
+     * @param apiCallFunction The function to call to get the next page.
+     * @param currentEntries  The current entries in the collection. Needed to determine the next page parameters.
+     * @param <TCollection>   The type of the collection to retrieve.
+     * @param <TEntries>      The type of the entries in the collection.
+     * @return The next page of the collection.
+     * @throws EndOfPaginationError If there are no more pages to retrieve.
+     */
+    public final <TCollection extends PaginatedCollection, TEntries extends EasyPostResource> TCollection getNextPage(
+            Function<Map<String, Object>, TCollection> apiCallFunction, List<TEntries> currentEntries)
+            throws EndOfPaginationError {
+        int defaultPageSize = 20;
```

Comment:
issue: Let's not set a default here and instead not send this if the user doesn't specify. Reason is some endpoints have different default page sizes sadly. Without sending it, we can keep it uniform with the docs and endpoints without needing to specify every endpoint default in the lib